learn/get-started/write-a-graphql-api-with-ballerina.md

```diff
@@ -14,49 +14,72 @@ redirect_from:
+## Understand the implementation
 
-Usually, a GraphQL endpoint is defined using a GraphQL schema. Some languages require the GraphQL schema to create a
-GraphQL service (schema-first approach) while others do not need the schema to create the service
-(code-first approach). Ballerina GraphQL package uses the latter. Therefore, you do not need the schema file to create
-your service, and instead, once you write the Ballerina service, the Ballerina GraphQL package will generate the schema.
+Usually, a GraphQL endpoint is defined using a GraphQL schema. Even though some languages require the GraphQL schema to create a GraphQL service (schema-first approach), Ballerina does not need the schema to create the service
+(code-first approach). Once you write the Ballerina service, the Ballerina GraphQL package will generate the schema.
 
-The GraphQL endpoint you are going to create in this guide will have two main operations, `Query` and `Mutation`. The
-`Query` type will be used to read the data in the data source, and the `Mutation` operation will be used to update the
-data in the data source.
+The GraphQL endpoint you will create in this guide will have two main operations, `query` and `mutation`. The `query` type will be used to read the data in the data source, and the `mutation` operation will be used to update the data in the data source.
 
-### The `Query` Type
-The `Query` type has two fields.
+### The `query` type
+The `query` type has two fields.
+* Can be identified by the `resource` keyword in the function description.
 * The `all` field - This field will return all the data in the data source as an array.
-
 * The `filter` field - This field will return the data filtered by the ISO Code of a country(`isoCode`).
 
-### The `Mutation` Type
-The `Mutation` type has a single field.
+### The `mutation` type
+The `mutation` type has a single field.
+* Can be identified by the `remote` keyword in the function description.
 * The `add` field - This field will add a given entry to the data source.
 
-## Create a data source for the project
+## Create the service package
+
+Ballerina uses packages to group code. You need to create a Ballerina package, generate the service code in the package, and write the business logic.
+
+>**Info:** For more information on Ballerina packages, see [Organize Ballerina code](/learn/organize-ballerina-code).
+
+In the terminal, execute the command below to create the Ballerina package for the API implementation.
+
+```bash
+$ bal new graphql
+```
 
-Before writing the GraphQL service, let's create a data source for the project. This will mimic a database that stores
-the data for the service. In this guide, you are going to use an in-memory table in Ballerina as the data source.
+You view the output below.
+
+```bash
+Created new package 'graphql' at graphql.
+```
+
+This creates a directory named `graphql` with the default module along with a sample code for the service as shown below.
```

Comment:
`graphql` is not a good name for the package.

Reply:
Updated the package name to `covid19`.